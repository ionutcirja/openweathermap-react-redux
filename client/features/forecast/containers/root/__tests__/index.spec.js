import React from 'react';
import { shallow } from 'enzyme';
import { Wrapper } from '..';

describe('Forecast container', () => {
	const propsToRender = {
	  actions: {
      forecastRequest: jest.fn(),
    },
  };

	describe('render', () => {
		it('should render a Forecast component', () => {
			const wrapper = shallow(<Wrapper {...propsToRender} />);
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('componentDidMount', () => {
    it('should call forecastRequest action', () => {
      shallow(<Wrapper {...propsToRender} />);
      expect(propsToRender.actions.forecastRequest).toHaveBeenCalledWith({ location: 'London' });
    });
  });
});
