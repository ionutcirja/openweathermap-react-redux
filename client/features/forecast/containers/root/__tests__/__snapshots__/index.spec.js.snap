// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast container render should render a Forecast component 1`] = `<WithTheme(Component) />`;
